Tasks/ANT/anttask.ts

```diff
@@ -125,80 +40,232 @@ function publishTestResults(publishJUnitResults, testResultsFiles: string) {
-    reportBuildFile = path.join(buildRootPath, reportBuildFileName);
-    var summaryFileName = "coverage.xml";
-    summaryFile = path.join(buildRootPath, reportDirectoryName);
-    summaryFile = path.join(summaryFile, summaryFileName);
-    var coberturaCCFile = path.join(buildRootPath, "cobertura.ser");
-    
-    // clean any previous reports.
-    tl.rmRF(coberturaCCFile, true);
-    tl.rmRF(reportDirectory, true);
-    tl.rmRF(reportBuildFile, true);
-
-    var buildProps: { [key: string]: string } = {};
-    buildProps['buildfile'] = antBuildFile;
-    buildProps['classfilter'] = classFilter
-    buildProps['classfilesdirectories'] = classFilesDirectories;
-    buildProps['sourcedirectories'] = sourceDirectories;
-    buildProps['summaryfile'] = summaryFileName;
-    buildProps['reportdirectory'] = reportDirectory;
-    buildProps['ccreporttask'] = ccReportTask
-    buildProps['reportbuildfile'] = reportBuildFile;
-    try {
-        var codeCoverageEnabler = new tl.CodeCoverageEnabler('Ant', ccTool);
-        codeCoverageEnabler.enableCodeCoverage(buildProps);
-        tl.debug("Code coverage is successfully enabled.");
+function processAntOutputLine(line) {
+    if (line == null) {
+        return;
     }
-    catch (Error) {
-        tl.warning("Enabling code coverage failed. Check the build logs for errors.");
+
+    let javacText = "[javac] ";
+    //[java] [javac] c:\path\to\file:100: error: error_msg
+    let compileErrorFileRegexWin = /^(\[java\])?\s*\[javac\]\s*([^:]:[^:]+):(\d+):\s*(.+)$/
+    //[java] [javac] /path/to/file:100: error: error_msg
+    let compileErrorFileRegexUnix = /^(\[java\])?\s*\[javac\]\s*([^:]+):(\d+):\s*(.+)$/
+    let compileErrorFileRegex = (isWindows) ? compileErrorFileRegexWin : compileErrorFileRegexUnix;
+
+    let severity = null;
+    if (line.indexOf(javacText) >= 0) {
+        // parse javac errors and warnings
+        let matches = compileErrorFileRegex.exec(line);
+        if (matches) {
+            let errorMessage = matches[4];
+            if (errorMessage) {
+                if (errorMessage.startsWith('warning:')) {
+                    severity = 'warning';
+                } else if (errorMessage.startsWith('error:')) {
+                    severity = 'error';
+                }
+            }
+
+            tl.command('task.issue', {
+                type: severity,
+                sourcepath: matches[2],
+                linenumber: matches[3],
+            }, matches[0]);
+        }
     }
 }
 
-function publishCodeCoverage(codeCoverageOpted: boolean) {
-    if (codeCoverageOpted) {
-        tl.debug("Collecting code coverage reports");
-        var antRunner = tl.createToolRunner(anttool);
-        antRunner.arg('-buildfile');
-        if (pathExistsAsFile(reportBuildFile)) {
-            antRunner.pathArg(reportBuildFile);
-            antRunner.arg(ccReportTask);
+
+async function doWork() {
+
+    function enableCodeCoverage() {
```

Comment:
why are these functions inside the async function? Is it because they need to read inputs?